esmvalcore/_recipe.py

```diff
@@ -352,71 +352,86 @@ def _get_default_settings(variable, config_user, derive=False):
 def _add_fxvar_keys(fx_var_dict, variable):
     """Add keys specific to fx variable to use get_input_filelist."""
     fx_variable = dict(variable)
+    fx_variable.update(fx_var_dict)
 
     # set variable names
     fx_variable['variable_group'] = fx_var_dict['short_name']
-    fx_variable['short_name'] = fx_var_dict['short_name']
 
-    # specificities of project
+    # add special ensemble for CMIP5 only
     if fx_variable['project'] == 'CMIP5':
-        fx_variable['mip'] = 'fx'
         fx_variable['ensemble'] = 'r0i0p0'
```

Comment:
yes, now `mip`s are automatically retrived at line 386 (`# Get all fx-related mips ('fx' always first, original mip last)`) - this is something @schlunma has innovated so we can have automation for cases when mip should be eg Oyr instead of Ofx - a very useful concoction :beer: